include/deal.II/multigrid/mg_transfer.templates.h

```diff
@@ -284,27 +432,14 @@ MGTransferPrebuilt<VectorType>::copy_from_mg_add
  OutVector                       &dst,
  const MGLevelObject<VectorType> &src) const
 {
-  // For non-DG: degrees of
-  // freedom in the refinement
-  // face may need special
-  // attention, since they belong
-  // to the coarse level, but
-  // have fine level basis
-  // functions
+  // For non-DG: degrees of freedom in the refinement face may need special
+  // attention, since they belong to the coarse level, but have fine level
+  // basis functions
   for (unsigned int level=0; level<mg_dof_handler.get_triangulation().n_global_levels(); ++level)
     {
-      typedef std::vector<std::pair<types::global_dof_index, types::global_dof_index> >::const_iterator dof_pair_iterator;
-
-      // First add all indices local to this process
-      for (dof_pair_iterator i= copy_indices[level].begin();
-           i != copy_indices[level].end(); ++i)
-        dst(i->first) += src[level](i->second);
-
-      // Do the same for the indices where the level index is local,
-      // but the global index is not
-      for (dof_pair_iterator i= copy_indices_level_mine[level].begin();
-           i != copy_indices_level_mine[level].end(); ++i)
-        dst(i->first) += src[level](i->second);
+      internal::copy_vector_from_mg(src[level], copy_indices[level],
+                                    copy_indices_level_mine[level],
+                                    level, true, additional_vector_data, dst);
     }
   dst.compress(VectorOperation::add);
```

Comment:
Good question. I thought it would not but on a second thought I think you are right - we need to insert a `zero_out_ghosts()` in the copy function to avoid possible problems. I'll include that in my refactoring.